### ISSUE TITLE
pioneer game crashes on game save

### ISSUE THREAD

**Issue by sagitter** [pioneerspacesim/pioneer#5570]
### Observed behaviour
pioneer (cd91559), compiled in Fedora 37, is crashing with

```
Stack trace of thread 80278:
#0  0x00007f89bf4afe5c __pthread_kill_implementation (libc.so.6 + 0x8ce5c)
#1  0x00007f89bf45fa76 raise (libc.so.6 + 0x3ca76)
#2  0x00007f89bf4497fc abort (libc.so.6 + 0x267fc)
#3  0x00007f89bf6d7e80 _ZSt21__glibcxx_assert_failPKciS0_S0_ (libstdc++.so.6 + 0xd7e80)
#4  0x0000556b8a73fef3 _ZN11PropertyMap8iteratorppEv (pioneer + 0x375ef3)
#5  0x0000556b8a74245a _ZN11PropertyMap10SaveToJsonERN8nlohmann10basic_jsonISt3mapSt6vectorNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEblmdSaNS0_14adl_serializerEEE (pioneer + 0x37845a)
#6  0x0000556b8a598c7a _ZN4Body10SaveToJsonERN8nlohmann10basic_jsonISt3mapSt6vectorNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEblmdSaNS0_14adl_serializerEEEP5Space (pioneer + 0x1cec7a)
#7  0x0000556b8a4fcc7f _ZN11TerrainBody10SaveToJsonERN8nlohmann10basic_jsonISt3mapSt6vectorNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEblmdSaNS0_14adl_serializerEEEP5Space (pioneer + 0x132c7f)
#8  0x0000556b8a5ab70f _ZN4Game6ToJsonERN8nlohmann10basic_jsonISt3mapSt6vectorNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEblmdSaNS0_14adl_serializerEEE (pioneer + 0x1e170f)
#9  0x0000556b8a5b0cde _ZN4Game8SaveGameERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEPS_ (pioneer + 0x1e6cde)
#10 0x0000556b8a65075f _ZL16l_game_save_gameP9lua_State.lto_priv.0 (pioneer + 0x28675f)
#11 0x0000556b8a6a96ba luaD_precall (pioneer + 0x2df6ba)
#12 0x0000556b8a6bd493 luaV_execute (pioneer + 0x2f3493)
#13 0x0000556b8a6abdd0 f_call.lto_priv.0 (pioneer + 0x2e1dd0)
#14 0x0000556b8a6a7014 luaD_rawrunprotected (pioneer + 0x2dd014)
#15 0x0000556b8a6aafd3 luaD_pcall (pioneer + 0x2e0fd3)
#16 0x0000556b8a77becd lua_pcallk.constprop.0 (pioneer + 0x3b1ecd)
#17 0x0000556b8a6af38a luaB_xpcall (pioneer + 0x2e538a)
#18 0x0000556b8a6a96ba luaD_precall (pioneer + 0x2df6ba)
#19 0x0000556b8a6bd3e3 luaV_execute (pioneer + 0x2f33e3)
#20 0x0000556b8a6abdd0 f_call.lto_priv.0 (pioneer + 0x2e1dd0)
#21 0x0000556b8a6a7014 luaD_rawrunprotected (pioneer + 0x2dd014)
#22 0x0000556b8a6aafd3 luaD_pcall (pioneer + 0x2e0fd3)
#23 0x0000556b8a77becd lua_pcallk.constprop.0 (pioneer + 0x3b1ecd)
#24 0x0000556b8a69e622 _Z21pi_lua_protected_callP9lua_Stateii (pioneer + 0x2d4622)
#25 0x0000556b8a5382cb _ZN5PiGui10RunHandlerEdNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE (pioneer + 0x16e2cb)
#26 0x0000556b8a49d999 _ZN8GameLoop6UpdateEf (pioneer + 0xd3999)
#27 0x0000556b8a45a27a _ZN11Application3RunEv (pioneer + 0x9027a)
#28 0x0000556b8a439bac main (pioneer + 0x6fbac)
#29 0x00007f89bf44a510 __libc_start_call_main (libc.so.6 + 0x27510)
#30 0x00007f89bf44a5c9 __libc_start_main@@GLIBC_2.34 (libc.so.6 + 0x275c9)
#31 0x0000556b8a447ac5 _start (pioneer + 0x7dac5)
```
Original bug ticket in Fedora: https://bugzilla.redhat.com/show_bug.cgi?id=2174394

**Comment by impaktor** [comment]
> pioneer (https://github.com/pioneerspacesim/pioneer/commit/cd91559f6e6ab1bf42a6270c0826e4ea5f2d3f29), compiled in Fedora 37, is crashing with

Doesn't the bug report say the opposite, that the game saves fine on this commit?

**Comment by hanecak** [comment]
> Doesn't the bug report say the opposite, that the game saves fine on this commit?

To clarify:

1. game built BY Fedora (e.g. the RPM) is crashing on my Fedora install (Fedora 37), stacktrace above is from pioneer-20230203-2.20230301gitcd91559.fc37.x86_64
2. game built ON my Fedora install (e.g. repo cloned and compiled) game works (cd91559)

Thus I've also included following in https://bugzilla.redhat.com/show_bug.cgi?id=2174394 :

> Hence the problem might be in how the game is compiled or packaged.

**Comment by impaktor** [comment]
So I assume there isn't any way for us to reproduce this crash, unless we're on a Fedora system?

**Comment by sagitter** [comment]
> So I assume there isn't any way for us to reproduce this crash, unless we're on a Fedora system?

If you have a Fedora 37 installed somewhere, it's enough installing Pioneer by Software Manager or by command line with `dnf install pioneer` and test it.


> To clarify:

    game built BY Fedora (e.g. the RPM) is crashing on my Fedora install (Fedora 37), stacktrace above is from pioneer-20230203-2.20230301gitcd91559.fc37.x86_64
    game built ON my Fedora install (e.g. repo cloned and compiled) game works (cd91559)


At this point, the crash looks like depending by how Pioneer is [built](https://kojipkgs.fedoraproject.org//packages/pioneer/20230203/2.20230301gitcd91559.fc37/data/logs/x86_64/build.log) in the packaging process of Fedora.
@hanecak, do you have a log of your Pioneer compile process?

**Comment by impaktor** [comment]
@sagitter If you haven't seen it, this issue is solved by #5574
